clients/go/admin/config.go

```diff
@@ -37,14 +37,16 @@ const (
 )
 
 type Config struct {
-	Endpoint              config.URL      `json:"endpoint" pflag:",For admin types, specify where the uri of the service is located."`
-	UseInsecureConnection bool            `json:"insecure" pflag:",Use insecure connection."`
-	InsecureSkipVerify    bool            `json:"insecureSkipVerify" pflag:",InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name. Caution : shouldn't be use for production usecases'"`
-	CACertFilePath        string          `json:"caCertFilePath" pflag:",Use specified certificate file to verify the admin server peer."`
-	MaxBackoffDelay       config.Duration `json:"maxBackoffDelay" pflag:",Max delay for grpc backoff"`
-	PerRetryTimeout       config.Duration `json:"perRetryTimeout" pflag:",gRPC per retry timeout"`
-	MaxRetries            int             `json:"maxRetries" pflag:",Max number of gRPC retries"`
-	AuthType              AuthType        `json:"authType" pflag:"-,Type of OAuth2 flow used for communicating with admin."`
+	Endpoint                  config.URL      `json:"endpoint" pflag:",For admin types, specify where the uri of the service is located."`
+	UseInsecureConnection     bool            `json:"insecure" pflag:",Use insecure connection."`
+	InsecureSkipVerify        bool            `json:"insecureSkipVerify" pflag:",InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name. Caution : shouldn't be use for production usecases'"`
+	CACertFilePath            string          `json:"caCertFilePath" pflag:",Use specified certificate file to verify the admin server peer."`
+	MaxBackoffDelay           config.Duration `json:"maxBackoffDelay" pflag:",Max delay for grpc backoff"`
+	PerRetryTimeout           config.Duration `json:"perRetryTimeout" pflag:",gRPC per retry timeout"`
+	MaxRetries                int             `json:"maxRetries" pflag:",Max number of gRPC retries"`
+	AuthType                  AuthType        `json:"authType" pflag:"-,Type of OAuth2 flow used for communicating with admin."`
+	RefreshTokenPreemptively  bool            `json:"refreshTokenPreemptively" pflag:",Preemptively refreshes auth token with jitter before it expires (client credential only)."`
+	PreemptiveRefreshDuration config.Duration `json:"preemptiveRefreshDuration" pflag:",Max duration between token refresh attempt and token expiry."`
```

Comment:
Can we change this to `TokenRefreshWindow` (or a better babe).. so if it's 0, it means we will refresh when token expires. If it's 1min, then we will refresh a min before it expires... Etc.

Then we can get rid of the two knobs...